### PR TITLE
Feature members only

### PR DESCRIPTION
**Pull request by Jarfsoft**
**Features added:** 
- Users model and controller
- Posts model and controller
- Layouts
- Sign in requirement

### CODE REVIEW

**Review by Srjnnnn — CHANGES_REQUESTED** [review]
## Status: Changes Required :recycle: 

Hello team! It's great you make it till here :smile: However before moving on you need to give us a working app to review. Below I'll give you some details about the problem.

1error

- [x]  After I set up the app, (clone, checkout, bundle, DB setup) I run the local server and I got the error above. Even you've configured your index page as posts#index it redirects directly users/sing_in! I believe the issue is you've used before_action: authenticate_user! for every action on the controller. However, to see the index page, there must be no authentication. It is a tautology to authenticate the user before their actions. :100: 

2error

- [x]  You can see the output from the server in the picture above. First, it is requesting the index action and it gets an unauthorized error after this request. Then it's trying to navigate the user sign_in page directly.

Please address the issues above and kindly fix them. :recycle: 
If you have any questions about the review, feel free to ask me via Slack. :100: 
_Eyüp Sercan UYGUR_

File: app/views/posts/index.html.erb
Comment (teekaytech) [inline app/views/posts/index.html.erb]: Kindly move this logic into the helper module. As much as possible, ensure that there's no logic in views.

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone.
However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x]  The application layout should have Sign in / Sign out links (So they are not manually repeated in every view) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/)).
- You probably did this, but the profiler is covering the links. Kindly do the needful.
- [x] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.
      **Resources**
- [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
- [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright
- Check my inline comment below.




* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by descholar-ceo** [comment]
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @Jarfsoft and @alvarorf  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 

- [x] As a guest you should see only post's text ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png) from Odin) => This means that if I am not logged in, I should see posts but not their authors

- [x] You have some logics in views: There shoud be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from `Pathwright`
Resources
    - [x] View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb) => To help you understand what to change: Make sure that there are no if conditions, or loops in views

Resources that can help you: 
:point_right: [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code/)
:point_right: [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Review by LamiaSristy — APPROVED** [review]
# Status: Approved. 💯 🥇 

Well done Team!! 👏 👏 👏 👏 

Your project is now approved. Congratulations on your hard work. 🛠️ 


You can merge the branch now, good luck with your next project 🚀.

Feel free to reach me at @lamiasristy 🙋

**Pull request merged**
